### ISSUE TITLE
Fix configuration inconsistency in app &  Kubernetes depoyment in '_run/single'

### ISSUE THREAD

**Issue by hydrogen18** [ovrclk/akash#838]
We did some investigation and found a few things

1. Once the akashd pod is running the container is up but you can't get to it, you just get a 502 Bad gateway response
2. The akash application is listening on `tcp://localhost:26657` not on the wildcard address
3. Tendermint is creating a default `config.toml` file since we aren't providing one in the container
4. Even specifying the command line flag for the TCP port for RPC doesn't seem to override it, this should be fixed since command line switches should always win.

We need to make our configuration consistent & document how we're doing it as well as recommended methods.

**Comment by hydrogen18** [comment]
Here's the result of my investigation on this so far

1. I played with setting environmental variables like `AKASH_RPC_LADDR` on my local machine and they don't do anything that I can see with. I think we may have to call `viper.SetEnvKeyReplacer` to get it to replace the dot character with underscores when checking the environmental variables for configuration
2. There appears to be a bug in cosmos SDK when loading the configuration for tendermint, explained below

In `cmd /akash/cmd/root.go` we install a `PersistenPreRunE` function into cobra which gets executed and calls `server.InterceptConfigsPreRunHandler` which is expected from the cosmos SDK. That function is defined in this file 

https://github.com/cosmos/cosmos-sdk/blob/9856327d95eba4c3ae2d15d26a76ad3ac17b20b8/server/util.go#L57

That function calls a private function called `interceptConfigs` which has the following section 

```
	rootDir := rootViper.GetString(flags.FlagHome)
	configPath := filepath.Join(rootDir, "config")
	configFile := filepath.Join(configPath, "config.toml")

	conf := tmcfg.DefaultConfig()

	if _, err := os.Stat(configFile); os.IsNotExist(err) {
		tmcfg.EnsureRoot(rootDir)

		if err = conf.ValidateBasic(); err != nil {
			return nil, fmt.Errorf("error in config file: %v", err)
		}

		conf.RPC.PprofListenAddress = "localhost:6060"
		conf.P2P.RecvRate = 5120000
		conf.P2P.SendRate = 5120000
		conf.Consensus.TimeoutCommit = 5 * time.Second
		tmcfg.WriteConfigFile(configFile, conf)
	} else {
		rootViper.SetConfigType("toml")
		rootViper.SetConfigName("config")
		rootViper.AddConfigPath(configPath)
		if err := rootViper.ReadInConfig(); err != nil {
			return nil, fmt.Errorf("failed to read in app.toml: %w", err)
		}

		if err := rootViper.Unmarshal(conf); err != nil {
			return nil, err
		}
	}
```
Note that `rootViper` is created in `InterceptConfigsPreRunHandler`, passed into this function and does not leave that scope.

In the code above if `config.toml` cannot be loaded it is created from defaults. This file does not exist presently in container when started in kubernetes. If the file does exist, the file is parsed by called `rootViper.ReadInConfig` and then `rootViper.Unmarshal(confg)` is called to update the tendermint configuration with whatever is in `rootViper`.

`rootViper` contains the entirety of the configuration set on the command line due to the fact that the flags are bound to it in `InterceptConfigsPreRunHandler`. So although this section of code appears to read and load a configuration file, it actually picks up the command line flags. However, if `config.toml` did not exist then the call to `rootViper.Unmarshal` never happens. This means the command line flags are never used to update the tendermint configuration.

I was able to confirm this by adding to our kubernetes startup script `touch /home/config/config.toml` to create an empty file before running `akash`. When I did that and set the command line flag I was able to get the process to listen on `0.0.0.0::22657`.  I could then access the process over HTTP using the Ingress declared in Kubernetes.

So to me this is a bug in Cosmos we need to fix or a bug in our usage of Cosmos.

**Comment by hydrogen18** [comment]
I did a local checkout of cosmos SDK and moved the call to `rootViper.Unmarshal` to being after the `if` / `else` loop. After that change, things work as expected with respect to command line switches taking priority over the configuration file.

Still doing further investigation on why env. vars. don't work

**Comment by hydrogen18** [comment]
As a test I shoved in this 

```
 rootViper.SetEnvPrefix("AKASH")
  rootViper.SetEnvKeyReplacer(strings.NewReplacer(".", "_", "-", "_"))
  rootViper.AutomaticEnv()
```

And the env vars work as expected now. So this one situation with `rootViper` is causing both issues.

**Comment by boz** [comment]
Was this fixed in #862 ?

**Comment by hydrogen18** [comment]
It looks like the fix for this was reverted in https://github.com/ovrclk/akash/commit/dd86936e7473d52e3f2681ddcf43907af80f0f95 because we are no longer using the fork I made of Cosmos
